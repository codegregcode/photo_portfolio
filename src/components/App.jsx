import React from 'react';
import { Route, Routes } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import Title from './Title';
import Index from './Index';

import Mar172024 from './journals/Mar172024';
import Apr272024 from './journals/Apr272024';

import '../styles/app.css';
import 'react-toastify/dist/ReactToastify.css';

const App = () => {
  return (
    <div className="app-container">
      <ToastContainer />
      <Routes>
        <Route path="/" element={<Title />} />
        <Route path="/index" element={<Index />} />
        <Route path="/index/a-test-a-start" element={<Mar172024 />} />
        <Route path="/index/porto" element={<Apr272024 />} />
      </Routes>
    </div>
  );
};

export default App;
