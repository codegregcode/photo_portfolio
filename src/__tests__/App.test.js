import React, { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../components/App';

describe('App', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['']}>
        <App />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
