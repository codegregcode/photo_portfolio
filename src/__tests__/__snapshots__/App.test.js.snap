// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <div
      class="title-component"
    >
      <button
        type="button"
      >
        enter
        <br />
        (warning flashing images)
      </button>
    </div>
  </div>
</DocumentFragment>
`;
